src/utils/add-corpus-storage-accounts/add-corpus-storage-account.py

```diff
@@ -0,0 +1,101 @@
+import argparse
+import json
+import uuid
+
+from azure.common.client_factory import get_client_from_cli_profile
+from azure.mgmt.eventgrid import EventGridManagementClient
+from azure.mgmt.eventgrid.models import EventSubscription
+from azure.mgmt.storage import StorageManagementClient
+from azure.mgmt.storage.models import (
+    AccessTier,
+    Kind,
+    Sku,
+    SkuName,
+    StorageAccountCreateParameters,
+)
+
+# This was generated randomly and should be preserved moving forwards
+STORAGE_GUID_NAMESPACE = uuid.UUID("f7eb528c-d849-4b81-9046-e7036f6203df")
+
+
+def get_base_event(
+    client: EventGridManagementClient, resource_group: str, location: str
+) -> EventSubscription:
+    for entry in client.event_subscriptions.list_regional_by_resource_group(
+        resource_group, location
+    ):
+        if (
+            entry.name == "onefuzz1"
+            and entry.type == "Microsoft.EventGrid/eventSubscriptions"
+            and entry.event_delivery_schema == "EventGridSchema"
+            and entry.destination.endpoint_type == "StorageQueue"
+            and entry.destination.queue_name == "file-changes"
+        ):
+            return entry
+
+    raise Exception("unable to find base eventgrid subscription")
+
+
+def add_event_grid(src_account_id: str, resource_group: str, location: str) -> None:
+    client = get_client_from_cli_profile(EventGridManagementClient)
+    base = get_base_event(client, resource_group, location)
+
+    event_subscription_info = EventSubscription(
+        destination=base.destination,
+        filter=base.filter,
+        retry_policy=base.retry_policy,
+    )
+
+    topic_id = uuid.uuid5(STORAGE_GUID_NAMESPACE, src_account_id).hex
+
+    result = client.event_subscriptions.create_or_update(
+        src_account_id, "corpus" + topic_id, event_subscription_info
+    ).result()
+    if result.provisioning_state != "Succeeded":
```

Comment:
They should a retry loop when the state is [Creating](https://docs.microsoft.com/en-us/python/api/azure-mgmt-eventgrid/azure.mgmt.eventgrid.models.eventsubscriptionprovisioningstate?view=azure-python)

Reply:
`.result()` polls until the results are either success or failure.  We don't need to check for creating here.